### ISSUE TITLE
Performance of `.stream()` ?

### ISSUE THREAD

**Issue by CodeMan99** [mrmlnc/fast-glob#147]
### Environment

  * OS Version: Ubuntu 18.01.1 LTS
  * Node.js Version: v8.14.1

### Actual behavior
`fast-glob.stream()` is slower and uses more memory than the async (standard) implementation.

### Expected behavior
Similar time and lower memory usage than the async (standard) implementations.

### Steps to reproduce

  1. Modify [src/benchmark/suites/async/glob-stream.ts](https://github.com/mrmlnc/fast-glob/blob/master/src/benchmark/suites/async/glob-stream.ts) to use `fast-glob.stream`.
  2. Run `npm run bench-async`

### Code sample

```typescript
import * as path from 'path';

import * as glob from '../../../index';

import * as utils from '../../utils';

import { IPartialOptions } from '../../../managers/options';

const options: IPartialOptions = {
	cwd: path.join(process.cwd(), process.env.BENCHMARK_CWD as string),
	unique: false
};

const timeStart = utils.timeStart();

const entries: string[] = [];

const stream = glob.stream(['**/*', '!**/*.txt'], options);

stream.on('data', (data: string) => {
	entries.push(data);
});
stream.on('error', () => {
	process.exit(0);
});
stream.once('end', () => {
	const memory = utils.getMemory();
	const time = utils.timeEnd(timeStart);
	const measures = utils.getMeasures(entries.length, time, memory);

	console.info(measures);
});
```

### Notes

It seems like there are some bad assumptions here. Would you mind looking into the performance here? Pretty low priority, I am not currently using `fast-glob.stream` in production.

**Comment by mrmlnc** [comment]
```
new — fast-glob@3
old — fast-glob@2

===> Benchmark pattern (*) with 100 launches (stream)

new-async.js
(TIME) 9.075ms ±0.696% | (MEMORY) 3.553MB ±0.011% | Entries: 7 | Errors: 0 | Retries: 1
new-stream.js
(TIME) 13.392ms ±1.110% | (MEMORY) 3.763MB ±0.010% | Entries: 7 | Errors: 0 | Retries: 1
old-async.js
(TIME) 13.400ms ±1.184% | (MEMORY) 6.084MB ±0.016% | Entries: 7 | Errors: 0 | Retries: 1
old-stream.js
(TIME) 16.348ms ±1.438% | (MEMORY) 6.177MB ±0.019% | Entries: 7 | Errors: 0 | Retries: 1
```

**Comment by mrmlnc** [comment]
Fixed by #196. This issue will be fixed with release `fast-glob@3`. You can check release issue #181.

**before**

```
(TIME) 359.030ms ±7.080% | (MEMORY) 6.691MB ±0.062% | Entries: 5211 | Errors: 0 | Retries: 1
```

**after**

```
(TIME) 245.260ms ±3.078% | (MEMORY) 5.811MB ±0.037% | Entries: 5211 | Errors: 0 | Retries: 1
```
